shared/src/commonMain/kotlin/io/customer/shared/tracking/queue/BackgroundQueue.kt

```diff
@@ -0,0 +1,204 @@
+package io.customer.shared.tracking.queue
+
+import io.customer.shared.Platform
+import io.customer.shared.common.CustomAttributes
+import io.customer.shared.database.TrackingTaskQueryHelper
+import io.customer.shared.sdk.meta.Workspace
+import io.customer.shared.tracking.api.model.Device
+import io.customer.shared.tracking.constant.MetricEvent
+import io.customer.shared.tracking.constant.TrackingType
+import io.customer.shared.tracking.model.Activity
+import io.customer.shared.tracking.model.Task
+import io.customer.shared.util.DateTimeUtil
+import io.customer.shared.util.Logger
+import io.customer.shared.work.JobDispatcher
+import io.customer.shared.work.JobExecutor
+import io.customer.shared.work.runOnBackground
+import io.customer.shared.work.runOnMain
+import kotlinx.coroutines.Job
+
+/**
+ * Background queue is responsible for queuing tasks to trigger when needed. It works asynchronously
+ * and does not blocks the caller for any operation. Caller should provide [TaskResultListener] to
+ * listen on results for any request.
+ */
+interface BackgroundQueue {
+    fun queueIdentifyProfile(
+        profileIdentifier: String,
+        attributes: CustomAttributes,
+        listener: TaskResultListener<Unit>? = null,
+    ): Job
+
+    fun queueTrack(
+        profileIdentifier: String?,
+        name: String,
+        trackingType: TrackingType,
+        attributes: CustomAttributes,
+        listener: TaskResultListener<Unit>? = null,
+    ): Job
+
+    fun queueRegisterDevice(
+        profileIdentifier: String?,
+        device: Device,
+        listener: TaskResultListener<Unit>? = null,
+    ): Job
+
+    fun queueDeletePushToken(
+        profileIdentifier: String?,
```

Comment:
Why is profile identifier optional?

Reply:
Responded [above](https://github.com/customerio/mobile-shared/pull/5#discussion_r1054401372)